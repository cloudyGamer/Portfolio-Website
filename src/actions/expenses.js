 /* 
  * To change this license header, choose License Headers in Project Properties.
  * To change this template file, choose Tools | Templates
  * and open the template in the editor.
  */
  import uuid from 'uuid';
 //ADD EXPENSE
 export const addExpense = ({
      description='',
      amount=0,
      note='',
      createdAt=0
 }={}
     ) => ({
      type: 'ADD_EXPENSE',
      expense : {
           id:uuid(),
           description,
           note,
           amount,
           createdAt
      }
      
});
export const removeExpense = ({id}={}) => ({
     type: 'REMOVE_EXPENSE',
     id
});
export const editExpense = (id,updates) => ({
 type: 'EDIT_EXPENSE',
      id,
      updates
 
});
