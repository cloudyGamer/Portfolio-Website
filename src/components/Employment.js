 /* 
  * To change this license header, choose License Headers in Project Properties.
  * To change this template file, choose Tools | Templates
  * and open the template in the editor.
  */
  
import React from 'react'; 
 const Employment = () => (
  <div>
  This is from my Employment  component
  </div>  

  );
 
 export default Employment;