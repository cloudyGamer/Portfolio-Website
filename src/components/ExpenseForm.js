 
import React from 'react'; 
import moment from 'moment';
import  {SingleDatePicker} from 'react-dates';
//import 'react-dates/lib/css/_datepicker.css';

const now = moment();
//console.log(now.format("MMM Do, YYYY"));

export default class ExpenseForm extends React.Component {
          constructor(props) {
               super(props);
               
               this.state = {
               description: props.expense ? props.expense.description : '',
               textArea: props.expense? props.expense.textArea:'',
               amount: props.expense ? (props.expense.amount/100).toString() : '',
               createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
               calendarFocused: false,
               error: ''
               }
               
          }
        
        
          onDescriptionChange = (e) => {
              const description = e.target.value;
              this.setState(() => ({ description }));
          };
          
          onTextAreaChange = (e) => {
               const textArea = e.target.value;
               this.setState(() => ({ textArea }));
          };
          
          onAmountChange = (e) => {
               const amount = e.target.value;
               if(!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)){
                    this.setState(() => ({ amount }));
               }
          };
          
          onDateChange = (createdAt) => {
               if (createdAt) {
                    this.setState(() => ({ createdAt }));
               };
          };
          
          onFocusChange = ({ focused }) => {
               this.setState(() => ({ calendarFocused: focused }));
          };
          
          onSubmit = (e) => {
               e.preventDefault();
               if (!this.state.amount||!this.state.description) {
                    this.setState(() => ({ error: "please provide description and amount" }));
                    //set error state equal to "please provide description and amount"
               }else {
                    this.setState(() => ({ error: '' }));
                    this.props.onSubmit({
                         description:this.state.description,
                         amount: parseFloat(this.state.amount,10)*100,
                         createdAt: this.state.createdAt.valueOf(),
                         textArea: this.state.textArea
                          
                    });
               }

          }
          
        
        render() {
        
        return(<div  xmlns="http://www.w3.org/1999/xhtml">
                    {this.state.error&&<p>{this.state.error}</p>}
                  <form onSubmit={this.onSubmit}>
                    <input 
                         type="text"
                         placeholder="Description"
                         autoFocus
                         value={this.state.description}
                         onChange={this.onDescriptionChange}
                    />
                    <input
                         type="text"
                         placeholder="Amount"
                         value={this.state.amount}
                         onChange={this.onAmountChange}
                    />
                    <SingleDatePicker
                         date={this.state.createdAt}
                         onDateChange = {this.onDateChange}
                         focused={this.state.calendarFocused}
                         onFocusChange={this.onFocusChange}
                         numberOfMonths={1}
                         isOutsideRange={() => false}/>
                    <textarea
                         placeholder='Add a note for your expense (optional)'
                         value={this.state.textArea}
                         onChange={this.onTextAreaChange}
                    />
                       {this.props.expense? <button>Edit Expense</button>:<button>Add Expense</button>}
                  </form>  
               </div>);
    }  
    
}

