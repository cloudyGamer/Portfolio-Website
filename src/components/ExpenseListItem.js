 /* 
  * To change this license header, choose License Headers in Project Properties.
  * To change this template file, choose Tools | Templates
  * and open the template in the editor.
  */
 
 import React from 'react';
  import { Link } from 'react-router-dom';
 //description //createdAt //amount
 //use map - the map goes in expense list. pass the individual items as props
 
 export const ExpenseListItem = ({dispatch, description, amount, createdAt, id}) => (
   <div xmlns="http://www.w3.org/1999/xhtml">
   <Link to={`/edit/${id}`}><h3>{description}</h3></Link>
     <p>{amount} - {createdAt}</p>

   </div>  

   );

export default ExpenseListItem;
