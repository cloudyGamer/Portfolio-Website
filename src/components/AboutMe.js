
import React from 'react'; 
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { addExpense } from '../actions/expenses';

export class AboutMe extends React.Component {
     onSubmit = (expense) => {
          this.props.addExpense(expense);
          this.props.history.push(`/`);   
     };
     
     render() {
     
       return (
          <div>
             <h1>About Me</h1>    
             <ExpenseForm 
               onSubmit={this.onSubmit}
               xmlns="http://www.w3.org/1999/xhtml"
             /> 
         </div>
      );
     }
};

const mapDispatchToProps = (dispatch) => ({
     addExpense: (expense) => dispatch(addExpense(expense))
});

 export default connect(undefined, mapDispatchToProps)(AboutMe);