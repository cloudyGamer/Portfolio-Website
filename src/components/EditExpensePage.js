 /* 
  * To change this license header, choose License Headers in Project Properties.
  * To change this template file, choose Tools | Templates
  * and open the template in the editor.
  */
 
import React from 'react'; 
import { connect } from 'react-redux';
import { editExpense,removeExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

export class EditExpensePage extends React.Component {
     onSubmit = (expense) => {
          console.log("updated",expense);
          this.props.editExpense(this.props.expense.id,expense);
          this.props.history.push(`/`);
     };
     
     removeExpense = () => {
          this.props.removeExpense({id: this.props.expense.id});
          this.props.history.push(`/`);
     };
     render(){
          return(
               <div>
                 <h1>Expense Form</h1>    
                    <ExpenseForm xmlns="http://www.w3.org/1999/xhtml"
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}
                    /> 
                    <button 
                    onClick={this.removeExpense}>Remove</button>
               </div>  
            );
     }
};
const mapDispatchToProps = (dispatch,props) => ({
     editExpense: (expenseid,expense) => dispatch(editExpense(expenseid,expense)),
     removeExpense: (expense) => dispatch(removeExpense(expense))
});

const mapStateToProps = (state,props) => {
     return {
          expense : state.expenses.find((expense) => expense.id===props.match.params.id)
     };
};

 export default connect(mapStateToProps,mapDispatchToProps)(EditExpensePage);