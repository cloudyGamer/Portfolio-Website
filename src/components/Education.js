import React from 'react'; 
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { addExpense } from '../actions/expenses';

export class Education extends React.Component {
     onSubmit = (expense) => {
          this.props.addExpense(expense);
          this.props.history.push(`/`);   
     };
     
     render() {
     
       return (
          <div>
             <h1>Education</h1>    
         </div>
      );
     }
};

const mapDispatchToProps = (dispatch) => ({
     addExpense: (expense) => dispatch(addExpense(expense))
});

 export default connect(undefined, mapDispatchToProps)(Education);