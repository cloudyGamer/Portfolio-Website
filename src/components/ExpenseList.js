 /* 
  * To change this license header, choose License Headers in Project Properties.
  * To change this template file, choose Tools | Templates
  * and open the template in the editor.
  */
 
 import React from 'react';
 import { connect } from 'react-redux';
 import ExpenseListItem from './ExpenseListItem';
 import selectExpenses from '../selectors/expenses';
 export const ExpenseList = (props) => (
   <div>
   {
   props.expenses.length === 0 ? (<p>no expenses</p>) : (
   props.expenses.map((expense) => {
        return <ExpenseListItem  
         {...expense}
         key={expense.id}
        />
        })
          )
          }
   </div>  

   );
const mapStateToProps = (state) => {
      return {
           expenses: selectExpenses(state.expenses, state.filters)
      };
 };
export default connect(mapStateToProps)(ExpenseList);