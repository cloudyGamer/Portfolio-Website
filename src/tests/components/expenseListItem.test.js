import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseListItem } from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('should render ExpenseListItem with message', () => {
     const wrapper= shallow(<ExpenseListItem 
     xmlns="http://www.w3.org/1999/xhtml" {...expenses[0]} key={expenses[0].id}
       />);
     expect(wrapper).toMatchSnapshot();
});
