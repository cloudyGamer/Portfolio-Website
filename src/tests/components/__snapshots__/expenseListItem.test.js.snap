// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with message 1`] = `
<div
  xmlns="http://www.w3.org/1999/xhtml"
>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      Gum
    </h3>
  </Link>
  <p>
    195
     - 
    0
  </p>
</div>
`;
